plugins/none-ls/servers.nix

```diff
@@ -376,6 +297,47 @@ in {
     enabledSources = builtins.filter (source: source.enable) flattenedSources;
   in
     mkIf cfg.enable {
+      # ASSERTIONS FOR DEVELOPMENT PURPOSES: Any failure should be caught by CI before deployment.
+      # Ensure that the keys of the manually declared `builtinPackages` match the ones from upstream.
+      assertions = let
+        upstreamToolNames = unique (
+          flatten
+          (
+            mapAttrsToList
+            (_: attrNames)
+            noneLsBuiltins
+          )
+        );
+        localToolNames = attrNames builtinPackages;
+
+        undeclaredToolNames =
+          filter
+          # Keep tool names which are not declared locally
+          (toolName: !(elem toolName localToolNames))
+          upstreamToolNames;
+
+        uselesslyDeclaredToolNames =
+          filter
+          # Keep tool names which are not in upstream
+          (toolName: !(elem toolName upstreamToolNames))
+          localToolNames;
+      in [
+        {
+          assertion = (length undeclaredToolNames) == 0;
+          message = ''
+            Nixvim (plugins.none-ls): Some tools from upstream are not declared locally in `builtinPackages`.
+            -> [${concatStringsSep ", " undeclaredToolNames}]
+          '';
```

Comment:
Could this be a warning, instead of an error assertion?
getting the following on current unstable-small:
```
error:
Failed assertions:
- Nixvim (plugins.none-ls): Some tools from upstream are not declared locally in `builtinPackages`.
-> [puppet_lint, findent, gleam_format]
```

Reply:
I'm currently getting a error even on normal unstable.

Reply:
Same here